src/plugins/discover/server/locator/query_from_locator.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { AggregateQuery, Query } from '@kbn/es-query';
+import { LocatorServicesDeps } from '.';
+import { DiscoverAppLocatorParams } from '../../common';
+
+/**
+ * @internal
+ */
+export const queryFromLocatorFactory = (services: LocatorServicesDeps) => {
+  /**
+   * @public
+   */
+  const queryFromLocator = async (
+    params: DiscoverAppLocatorParams
+  ): Promise<Query | AggregateQuery | undefined> => {
+    return params.query;
+
+    // TODO: support query from saved search
```

Comment:
What would need to happen for this TODO to complete?

Reply:
if would need to take params.savedObjectId, fetch the saved object and take the query from there. 
I didn't do it because we need to decide which query takes precedence (I'd expect the params.query) and this as logic would work differently compared to csv_searchsource I didn't want Discover UI to have different behavior for non-ESQL and ESQL export. I think best would be to clean it up here https://github.com/elastic/kibana/issues/151190